saleor/checkout/tests/test_actions.py

```diff
@@ -77,6 +166,116 @@ def test_transaction_amounts_for_checkout_updated_with_already_fully_paid(
+        transaction, manager=plugins_manager, user=None, app=app
+    )
+
+    # then
+    flush_post_commit_hooks()
+    checkout.refresh_from_db()
+    assert checkout.charge_status == CheckoutChargeStatus.FULL
+    assert checkout.authorize_status == CheckoutAuthorizeStatus.FULL
+    mocked_fully_paid.assert_called_with(checkout, webhooks=set())
+    mocked_automatic_checkout_completion_task.assert_called_once_with(
+        checkout.pk, None, app.id
+    )
+
+
+@patch("saleor.checkout.tasks.automatic_checkout_completion_task.delay")
+@patch("saleor.plugins.manager.PluginsManager.checkout_fully_paid")
+def test_transaction_amounts_for_checkout_updated_fully_authorized(
+    mocked_fully_paid,
+    mocked_automatic_checkout_completion_task,
+    checkout_with_items,
+    transaction_item_generator,
+    plugins_manager,
+):
+    # given
+    checkout = checkout_with_items
+    lines, _ = fetch_checkout_lines(checkout)
+    checkout_info = fetch_checkout_info(checkout, lines, plugins_manager)
+    checkout_info, _ = fetch_checkout_data(checkout_info, plugins_manager, lines)
+    transaction = transaction_item_generator(
+        checkout_id=checkout.pk,
+        authorized_value=checkout_info.checkout.total.gross.amount,
+    )
+    assert checkout_info.channel.automatically_complete_fully_paid_checkouts is False
+
+    # when
+    transaction_amounts_for_checkout_updated(
+        transaction, manager=plugins_manager, user=None, app=None
+    )
+
+    # then
+    flush_post_commit_hooks()
+    checkout.refresh_from_db()
+    assert checkout.charge_status == CheckoutChargeStatus.NONE
+    assert checkout.authorize_status == CheckoutAuthorizeStatus.FULL
+    assert not mocked_fully_paid.called
+    assert not mocked_automatic_checkout_completion_task.called
+
+
+@patch("saleor.checkout.tasks.automatic_checkout_completion_task.delay")
+@patch("saleor.plugins.manager.PluginsManager.checkout_fully_paid")
+def test_transaction_amounts_for_checkout_fully_authorized_automatic_checkout_complete(
+    mocked_fully_paid,
+    mocked_automatic_checkout_completion_task,
+    checkout_with_items,
+    transaction_item_generator,
+    plugins_manager,
+    staff_user,
+):
+    # given
+    checkout = checkout_with_items
+    lines, _ = fetch_checkout_lines(checkout)
+    checkout_info = fetch_checkout_info(checkout, lines, plugins_manager)
+    checkout_info, _ = fetch_checkout_data(checkout_info, plugins_manager, lines)
+    transaction = transaction_item_generator(
+        checkout_id=checkout.pk,
+        authorized_value=checkout_info.checkout.total.gross.amount,
+    )
+    channel = checkout_info.channel
+    channel.automatically_complete_fully_paid_checkouts = True
+    channel.save(update_fields=["automatically_complete_fully_paid_checkouts"])
+
+    # when
+    transaction_amounts_for_checkout_updated(
+        transaction, manager=plugins_manager, user=staff_user, app=None
+    )
+
+    # then
+    flush_post_commit_hooks()
```

Comment:
I would recommend to not using `flush_post_commit_hooks` in new tests. In case of failure, the DB is not clean up correctly and it causes that next tests are also failing. Pytest-django has dedicated fixture for handling on_commit actions [ref](https://pytest-django.readthedocs.io/en/latest/helpers.html#django-capture-on-commit-callbacks). 
```python
with django_capture_on_commit_callbacks(execute=True) as callbacks:
    transaction_amounts_for_checkout_updated(
        transaction, manager=plugins_manager, user=staff_user, app=None
    )
```
(We also have a task to fully drop the usage of `flush_post_commit_hooks()`)